src/Symfony/Component/Lock/Tests/Store/ZookeeperStoreTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Tests\Store;
+
+/*
+ *
+ * @author Ganesh Chandrasekaran <gchandrasekaran@wayfair.com>
+ *
+ * @requires extension zookeeper
+ */
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\Store\ZookeeperStore;
+use Throwable;
+use Zookeeper;
+
+class ZookeeperStoreTest extends AbstractStoreTest
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function setupBeforeClass()
+    {
+        $zookeeper_server = getenv('ZOOKEEPER_HOST').':2181';
+
+        try {
+            $zookeeper = new Zookeeper(implode(',', array($zookeeper_server)));
+            // The session creation happens asynchronously, so we can wait for a second to make sure it is established.
+            sleep(1);
```

Comment:
perhaps a small while loop and check every 10ms is sufficient? Should probably try to avoid long sleeps.

Reply:
Have added a while loop of `10 ms` each but, still keeping the max sleep time to `1 second`